### ISSUE TITLE
`node download_binaries.js` command fails

### ISSUE THREAD

**Issue by 0x1mason** [RuntimeTools/appmetrics#218]
When installing from NPM, the `install` script fails. The problem is `install` is executed before the dependencies in `download_binaries.js` are installed. This is in Node 0.12 and npm 2.14. Installing all the dependencies in the install script may fix the problem.

**Comment by 0x1mason** [comment]
Something is looking for `json-stringify-safe`. Installing it as a peer fixes the issue.

**Comment by tobespc** [comment]
Are you installing from npm registry or from github as the download_binaries.js script gets replaced in the package we put up on https://www.npmjs.com/package/appmetrics and is called download_all_binaries.js

If you uninstall your json-stringify-safe module and appmetrics, and then attempt to reinstall it, can you paste in the trace showing the error please ?

Thanks

**Comment by 0x1mason** [comment]
From the NPM registry. I got another failure which is the same as the `json-stringify-safe`.
```shell
> appmetrics@1.0.11 install /home/blah/workspace/mccp-test-run/node_modules/appmetrics-elk/node_modules/appmetrics
> node download_all_binaries.js

module.js:338
    throw err;
          ^
Error: Cannot find module 'readable-stream/duplex'
    at Function.Module._resolveFilename (module.js:336:15)
    at Function.Module._load (module.js:278:25)
    at Module.require (module.js:365:17)
    at require (module.js:384:17)
    at Object.<anonymous> (/home/blah/workspace/mccp-test-run/node_modules/request/node_modules/bl/bl.js:1:82)
    at Module._compile (module.js:460:26)
    at Object.Module._extensions..js (module.js:478:10)
    at Module.load (module.js:355:32)
    at Function.Module._load (module.js:310:12)
    at Module.require (module.js:365:17)
npm ERR! Linux 3.13.0-43-generic
npm ERR! argv "/usr/bin/node" "/usr/bin/npm" "install"
npm ERR! node v0.12.9
npm ERR! npm  v2.14.9
npm ERR! code ELIFECYCLE
```

**Comment by tobespc** [comment]
the next release of appmetrics is going to remove the need to pull binaries from a separate host as all binaries will be hosted on npm

**Comment by sam-github** [comment]
@tobespc our CI is failing, is it possible to get a patch out that has dependencies on the modules used by the current install script?

**Comment by tobespc** [comment]
@sam-github  I'm on it :-)

**Comment by tobespc** [comment]
are you installing the latest 1.0.12 version ? Went live 2 days go ?

**Comment by tobespc** [comment]
this looks to be the same as https://github.com/RuntimeTools/appmetrics/issues/169

**Comment by tobespc** [comment]
Does upgrading to npm3 resolve it ?

**Comment by 0x1mason** [comment]
Thanks.

**Comment by 0x1mason** [comment]
Unfortunately, our workaround was to remove appmetrics from our production deployments. We still use it in test. Perhaps we will retry with the newer version at a later date.

**Comment by sjanuary** [comment]
I think this has been fixed by #289
